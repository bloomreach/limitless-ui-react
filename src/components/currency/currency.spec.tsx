import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import { Currency } from './';

describe('Currency', () => {
  it('should render successfully in different currencies and locales', () => {
    const currencies = ['USD', 'EUR', 'GBP', 'INR', 'AED', 'CNY', 'JPY'];
    const locales = ['en-US', 'nl-NL', 'sk-SK', 'ja-JP', 'hi-IN', 'ar-AE', 'zh-CN'];
    const { container: { firstElementChild } } = render(
      <div>
        {locales.map(locale => {
            return currencies.map(currency => {
              return (<Currency key={`${locale}-${currency}`} currency={currency} locale={locale} />);
            });
        })}
      </div>
    );

    expect(firstElementChild).toMatchSnapshot();
  });
});
