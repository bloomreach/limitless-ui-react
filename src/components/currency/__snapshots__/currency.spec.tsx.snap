// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Currency > should render successfully in different currencies and locales 1`] = `
<div>
  <span
    class="lui-price"
  >
    $
  </span>
  <span
    class="lui-price"
  >
    €
  </span>
  <span
    class="lui-price"
  >
    £
  </span>
  <span
    class="lui-price"
  >
    ₹
  </span>
  <span
    class="lui-price"
  >
    AED
  </span>
  <span
    class="lui-price"
  >
    CN¥
  </span>
  <span
    class="lui-price"
  >
    ¥
  </span>
  <span
    class="lui-price"
  >
    US$
  </span>
  <span
    class="lui-price"
  >
    €
  </span>
  <span
    class="lui-price"
  >
    £
  </span>
  <span
    class="lui-price"
  >
    ₹
  </span>
  <span
    class="lui-price"
  >
    AED
  </span>
  <span
    class="lui-price"
  >
    CN¥
  </span>
  <span
    class="lui-price"
  >
    JP¥
  </span>
  <span
    class="lui-price"
  >
    USD
  </span>
  <span
    class="lui-price"
  >
    €
  </span>
  <span
    class="lui-price"
  >
    GBP
  </span>
  <span
    class="lui-price"
  >
    INR
  </span>
  <span
    class="lui-price"
  >
    AED
  </span>
  <span
    class="lui-price"
  >
    CNY
  </span>
  <span
    class="lui-price"
  >
    JPY
  </span>
  <span
    class="lui-price"
  >
    $
  </span>
  <span
    class="lui-price"
  >
    €
  </span>
  <span
    class="lui-price"
  >
    £
  </span>
  <span
    class="lui-price"
  >
    ₹
  </span>
  <span
    class="lui-price"
  >
    AED
  </span>
  <span
    class="lui-price"
  >
    元
  </span>
  <span
    class="lui-price"
  >
    ￥
  </span>
  <span
    class="lui-price"
  >
    $
  </span>
  <span
    class="lui-price"
  >
    €
  </span>
  <span
    class="lui-price"
  >
    £
  </span>
  <span
    class="lui-price"
  >
    ₹
  </span>
  <span
    class="lui-price"
  >
    AED
  </span>
  <span
    class="lui-price"
  >
    CN¥
  </span>
  <span
    class="lui-price"
  >
    JP¥
  </span>
  <span
    class="lui-price"
  >
    ‏US$
  </span>
  <span
    class="lui-price"
  >
    ‏€
  </span>
  <span
    class="lui-price"
  >
    ‏UK£
  </span>
  <span
    class="lui-price"
  >
    ‏₹
  </span>
  <span
    class="lui-price"
  >
    ‏دإ‏
  </span>
  <span
    class="lui-price"
  >
    ‏CN¥
  </span>
  <span
    class="lui-price"
  >
    ‏JP¥
  </span>
  <span
    class="lui-price"
  >
    US$
  </span>
  <span
    class="lui-price"
  >
    €
  </span>
  <span
    class="lui-price"
  >
    £
  </span>
  <span
    class="lui-price"
  >
    ₹
  </span>
  <span
    class="lui-price"
  >
    AED
  </span>
  <span
    class="lui-price"
  >
    ¥
  </span>
  <span
    class="lui-price"
  >
    JP¥
  </span>
</div>
`;
